### PR TITLE
Add support for EXAT in SET command

### PR DESCRIPTION
**Pull request by colommar**
### Description

This pull request adds support for the `EXAT` option to the `SET` command, similar to the `EXAT` option in Redis. The `EXAT` option sets the specified Unix time at which the key will expire, in seconds.
Solving the issue #202

### Changes Made

1. **Implemented `EXAT` Option**:
   - Updated the `SET` command to handle the `EXAT` option.
   - Parsed the Unix time provided and set the key to expire at the specified time.

2. **Unit Tests**:
   - Added unit tests for the `EXAT` option to ensure correct behavior.
   - Verified that the key expires at the correct time when using `EXAT`.

3. **Integration Tests**:
   - Ported relevant integration tests from the Redis test suite in `aof_test.go`

4. **Benchmarking**:
   - Used `benchmem` to track memory allocations and optimize where necessary.

### CODE REVIEW

**Comment by arpitbbhayani** [comment]
Hello @colommar,

This is pretty amazing PR. Thanks!
I see there are changes in go.mod file, any specific reasons for it? Also your changes led to Merge Conflicts, can you resolve them and everything looks good to be merged.

File: core/eval.go [hunk 1]
```diff
@@ -87,6 +88,25 @@ func evalSET(args []string) []byte {
...
+			}
+
+			exAt, err := strconv.ParseInt(args[i], 10, 64)
+			if err != nil {
+				return Encode(errors.New("ERR value is not an integer or out of range"), false)
+			}
+
+			currentTime := time.Now().Unix()
+			if exAt <= currentTime {
+				return Encode(errors.New("ERR invalid EXAT time in the past"), false)
```
Comment (lucifercr07) [inline core/eval.go:105]: This behaviour doesn't reflect the same what's done in redis. 
Redis does not reject the `EXAT` value if it is in the past. Instead, it sets the key and immediately expires it if the timestamp has already passed.
@JyotinderSingh @arpitbbhayani please correct me if I'm wrong here.

File: core/eval.go [hunk 2]
```diff
@@ -87,6 +88,25 @@ func evalSET(args []string) []byte {
...
 			}
 			exDurationMs = exDurationSec * 1000
+
+		case "EXAT", "exat":
+			i++
+			if i == len(args) {
+				return Encode(errors.New("ERR syntax error"), false)
+			}
+
+			exAt, err := strconv.ParseInt(args[i], 10, 64)
```
Comment (lucifercr07) [inline core/eval.go:98]: We also need to add positive integer check.

File: core/eval.go [hunk 3]
```diff
@@ -96,6 +116,7 @@ func evalSET(args []string) []byte {
 				return RESP_NIL
 			}
 
+
```
Comment (lucifercr07) [inline core/eval.go:119]: Nit: Please remove if unintentional

**Review by lucifercr07 — CHANGES_REQUESTED** [review]

**Comment by arpitbbhayani** [comment]
Thanks @colommar Changes look good :) Merging them.

**Pull request merged**